docs/en/introduction/architecture.md

```diff
@@ -9,43 +9,43 @@ The JuiceFS file system consists of three parts:
+JuiceFS supports a variety of common databases as the metadata engine, like Redis, TiKV, MySQL/MariaDB, PostgreSQL, and SQLite, and the list is still expanding. [Submit an issue](https://github.com/juicedata/juicefs/issues) if your favorite database is not supported.
 
-## How JuiceFS Stores Files {#how-juicefs-store-files}
+## How JuiceFS stores files {#how-juicefs-store-files}
 
-Traditional file systems use local disks to store both file data and metadata, JuiceFS formats data first and then stores them in object storage, with the corresponding metadata being stored in the metadata engine.
+Traditional file systems use local disks to store both file data and metadata. However, JuiceFS formats data first and then stores it in the object storage, with the corresponding metadata being stored in the metadata engine.
 
-Files are divided into 64M "Chunk"s, this allows for fast lookup using offsets, bringing better performance for large files. As long as file size doesn't change, no matter how many writes have been seen, chunk division stays the same. But note that chunk is invented to speedup data lookups, write is conducted on the "Slice" data structure.
+Files are divided into 64 MB *chunks*. This allows for fast lookup using offsets, bringing better performance for large files. The division of chunks remains consistent as long as the file size remains unchanged, regardless of the number of writes. But note that chunks are invented to speed up data lookups, while writes are executed on the *slice* data structure.
 
-"Slice" is designed for writes, with a maximum size of 64M. A slice represents one single continuous write, and a slice always belongs to a chunk, and cannot overlap between two adjacent chunks. Continuous write will produce a single slice, while random writes or slow append writes can form multiple slices inside a chunk. Arrangement of these slices are controlled by write patterns, they may be close to each other, overlap, or have empty space between them. Slices are persisted after `flush` calls, `flush` can be explicitly invoked by user, but if not, JuiceFS Client will automatically carry out `flush` to avoid buffer being written to full.
+A slice represents a single continuous write and has a maximum size of 64 MB. Each slice belongs to a chunk and cannot overlap between adjacent chunks. Continuous writes produce a single slice, while random writes or slow append writes can result in multiple slices inside a chunk. The arrangement of these slices is controlled by write patterns, which may be close to each other, overlap, or have empty space between them. Slices are persisted after `flush` calls. `flush` can be explicitly invoked by the user; otherwise, the JuiceFS client automatically performs `flush` to prevent buffer overflow.
```

Comment:
Not sure whether "prevent buffer overflow" is correct.

Reply:
it's fine (I wrote this section)